File: Questions/69. sqrt(x)/main.py
def my_sqrt(x: int) -> int:
    left, right = 0, x
    while left < right:
        mid = (left + right + 1) // 2
        if mid ** 2 > x:
            right = mid - 1
        else:
            left = mid

    return left

print(my_sqrt(9))
print(my_sqrt(8))
print(my_sqrt(-1))
print(my_sqrt(0))

File: Questions/69. sqrt(x)/main.cpp
#include<iostream>
using namespace std;
int mySqrt(int x) {
    int left = 0, right = x / 2 + 1;
    while (left < right) {
        int mid = left + (right - left + 1) / 2;
        if (mid > x / mid) {
            right = mid - 1;
        } else {
            left = mid;
        }
    }
    return left;
}

int main() {

    cout << mySqrt(2) << endl;
    cout << mySqrt(8) << endl;
    cout << mySqrt(0) << endl; 
    return 0;
}